### ISSUE TITLE
Deep link protocol

### ISSUE THREAD

**Issue by shocklateboy92** [mormaer/Mlem#380]
Hey, I'm putting together a link redirection service for Lemmy:
https://lemmy.guide

It lets people create links to that will open in their instance (set in browser local storage) in app of their choice.

At the moment, I only have Liftoff (and browser, obviously) in there. But I'd love to add more, including Mlem.
Does mlem have a protocol it's registered to open like say, `mlem://lemmy.world/c/programmer_humor@programming.dev`?

**Comment by tht7** [comment]
Guys, I would love it if we could open a ticket about this. It’s extremely interesting.

(feel free to assign it to me if no one’s interested)

**Comment by EricBAndrews** [comment]
Yeah, this is a *super* neat idea. Any estimate how much dev work we're talking about to make this happen?

**Comment by JakeShirley** [comment]
Conceptually it shouldn't be too hard to register a protocol and parse the incoming string.

**Comment by Sjmarf** [comment]
> Yeah, this is a *super* neat idea. Any estimate how much dev work we're talking about to make this happen?

Here's the documentation on this particular feature https://developer.apple.com/documentation/xcode/defining-a-custom-url-scheme-for-your-app

**Comment by tht7** [comment]
We already have most of it done (all the URL parsing and display stuff is done when I’ve done the in-app links opening)
Now we just need to add the actual thing

I would love to handle this today

**Comment by Sjmarf** [comment]
I really like the sound of this idea, by the way. I've found it annoying that links to a post don't always open in my own instance. Great job OP! I'll be using this in future. 

Maybe we could even make links shared from within Mlem use lemmy.guide? Or perhaps a setting for it?
